package/MDAnalysis/topology/ITPParser.py

```diff
@@ -265,10 +265,9 @@ def find_path(self, path):
             current_file = self.current_file
 
         try:
-            path = path.name
+            path = os.path.abspath(path.name)
```

Comment:
This try/except is to check whether `path` is a string (which triggers the AttributeError) or a `Path` object from `pathlib` (so Path.path is the string version). Does your fix still work if a relative string path is passed in instead, triggering the AttributeError?

By the way, your test should check for both, just in case :)

Reply:
Hi @lilyminium, that's a good observation! However, if the user isn't providing a string and triggering this `AttributeError`, it's probably doing the right thing by just passing and eventually raising an IO error. I can write a test to check if an IOerror is raised when a relative string is passed but not as a string.

Reply:
Hi @aditya-kamath, 

> if the user isn't providing a string and triggering this AttributeError

If a string is provided and the AttributeError is triggered, does your fix still work? The IOError test is probably not necessary. However, your test should check that your fix applies to both a relative `Path` (https://docs.python.org/3/library/pathlib.html) and a relative string.

> (so Path.path is the string version)

This was a typo, sorry. `Path.name` returns the base name of the Path, as a string.

Reply:
Hmm, the answer to if it will work when there's an Attribute error is no! :( 
I'm assuming that the situation when this will happen is when os.path.abspath is not able to convert a relative string to an absolute path. I'll write the two situations in my test to see if there's a situation when this might happen! :D